runtime/src/bank/builtins/core_bpf_migration/builtin.rs

```diff
@@ -0,0 +1,260 @@
+use {
+    super::{error::CoreBpfMigrationError, CoreBpfMigration},
+    crate::bank::Bank,
+    solana_sdk::{
+        account::{Account, AccountSharedData},
+        bpf_loader_upgradeable::get_program_data_address,
+        native_loader::ID as NATIVE_LOADER_ID,
+        pubkey::Pubkey,
+    },
+};
+
+/// Used to validate a built-in program's account before migrating to Core BPF.
+#[derive(Debug)]
+pub(crate) struct BuiltinConfig {
```

Comment:
Nit: same here, how about `BuiltinProgram` or `BuiltinProgramData`?

Reply:
In order to not confuse anyone with the other types that use `Builtin` in the name, and following the pattern with `SourceProgramBpfUpgradeable`, I've opted for `TargetBuiltinProgram` as the revised name here.